tests/end_to_end/end_to_end.cpp

```diff
@@ -2723,8 +2723,7 @@ TEST_CASE("load_native_program_negative8", "[end-to-end]")
 
     // Try to load the programs from the module with 0 programs.
     REQUIRE(
-        test_ioctl_load_native_programs(&provider_module_id, nullptr, 1, &map_handles, 1, &program_handles) ==
-        ERROR_INVALID_PARAMETER);
+        test_ioctl_load_native_programs(&provider_module_id, 1, &map_handles, 0, &program_handles) == ERROR_SUCCESS);
```

Comment:
nit: maybe we can also check that the map was actually created.

Reply:
Done (checked handle)